### PR TITLE
FIX (Extension) @W-16733527@ Additional telemetry for apex guru

### PR DESCRIPTION
**Pull request by jag-j**
This gets the following telemetry:
Number of antipatterns count
Number of violations with suggestions
Number of quick-fix accepted

### CODE REVIEW

File: src/apexguru/apex-guru-service.ts [hunk 1]
```diff
@@ -57,7 +57,9 @@ export async function runApexGuruOnFile(selection: vscode.Uri, runInfo: RunInfo)
 			new DiagnosticManager().displayDiagnostics([selection.fsPath], [ruleResult], diagnosticCollection);
 			TelemetryService.sendCommandEvent(Constants.TELEM_SUCCESSFUL_APEX_GURU_FILE_ANALYSIS, {
 				executedCommand: commandName,
-				duration: (Date.now() - startTime).toString()
+				duration: (Date.now() - startTime).toString(),
+				violationsCount: ruleResult.violations.length.toString(),
+				violationsWithSuggestedCode: getViolationsWithSuggestions(ruleResult).toString()
```
Comment (jfeingold35) [inline src/apexguru/apex-guru-service.ts:62]: Should this be `violationsWithSuggestedCodeCount` to match `violationsCount`?

File: src/lib/fixer.ts [hunk 2]
```diff
@@ -111,14 +111,14 @@ export class _ApexGuruFixGenerator extends FixGenerator {
     
         const action = new vscode.CodeAction(messages.fixer.fixWithApexGuruSuggestions, vscode.CodeActionKind.QuickFix);
         action.diagnostics = [this.diagnostic];
-
-        const edit = new vscode.WorkspaceEdit();
         const range = this.diagnostic.range;  // Assuming the range is the location of the existing code in the document
         const diagnosticStartLine = new vscode.Position(range.start.line, range.start.character);
-        edit.insert(document.uri, diagnosticStartLine, suggestedCode + '\n');
```
Comment (jag-j) [inline src/lib/fixer.ts:118]: If I do this, there is no way to get the telemetry on how many times users clicked on the "include apex guru suggestions" quick-fix. I had to to make this a command and add it to the action and I can report telemetry on that command.

File: src/extension.ts [hunk 3]
```diff
@@ -143,7 +143,16 @@ export async function activate(context: vscode.ExtensionContext): Promise<vscode
 					outputChannel: outputChannel
 				});
 		});
-		context.subscriptions.push(runApexGuruOnSelectedFile, runApexGuruOnCurrentFile);
+		const insertApexGuruSuggestions = vscode.commands.registerCommand(Constants.COMMAND_INCLUDE_APEX_GURU_SUGGESTIONS, async (document: vscode.TextDocument, position: vscode.Position, suggestedCode: string) => {
+			const edit = new vscode.WorkspaceEdit();
+			edit.insert(document.uri, position, suggestedCode);
+			await vscode.workspace.applyEdit(edit);
+			TelemetryService.sendCommandEvent(Constants.TELEM_SUCCESSFUL_APEX_GURU_FILE_ANALYSIS, {
```
Comment (jag-j) [inline src/extension.ts:150]: New telemetry to get number of times users used the suggestions ApexGuru provided, along with the number of lines that were included.

File: src/apexguru/apex-guru-service.ts (hunk 1 above)
Comment (jag-j) [reply src/apexguru/apex-guru-service.ts:62]: Yes! that sounds better. I'll rename it.

**Review by jfeingold35 — APPROVED** [review]

**Pull request merged**